Sources/Identity/CustomerInfoManager.swift

```diff
@@ -279,20 +284,89 @@ class CustomerInfoManager {
+                                     completion: completion)
+        }
+    }
+
+}
+
+// MARK: -
+
+private extension CustomerInfoManager {
+
+    func getCustomerInfo(appUserID: String,
+                         isAppBackgrounded: Bool,
+                         completion: @escaping CustomerAPI.CustomerInfoResponseHandler) {
+        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+            _ = Task<Void, Never> {
+                // Note: this is only able to post a single transaction,
+                // it can be improved in the future once `PostReceiptOperation` accepts multiple ones.
+                if let transaction = await self.transactionFetcher.unfinishedVerifiedTransactions.first {
+                    Logger.debug(Strings.customerInfo.posting_transaction_in_lieu_of_fetching_customerinfo(transaction))
+
+                    self.transactionPoster.handlePurchasedTransaction(
+                        transaction,
+                        data: .init(appUserID: appUserID,
+                                    presentedOfferingID: nil,
+                                    unsyncedAttributes: [:],
+                                    storefront: await Storefront.currentStorefront,
+                                    source: Self.sourceForUnfinishedTransaction),
+                        completion: completion
+                    )
+                } else {
+                    self.requestCustomerInfo(appUserID: appUserID,
+                                             isAppBackgrounded: isAppBackgrounded,
+                                             completion: completion)
+                }
+            }
+        } else {
+            return self.requestCustomerInfo(appUserID: appUserID,
+                                            isAppBackgrounded: isAppBackgrounded,
+                                            completion: completion)
         }
     }
 
+    private func requestCustomerInfo(appUserID: String,
+                                     isAppBackgrounded: Bool,
+                                     completion: @escaping CustomerAPI.CustomerInfoResponseHandler) {
+        let allowComputingOffline = self.offlineEntitlementsManager.shouldComputeOfflineCustomerInfo(
+            appUserID: appUserID
+        )
+
+        self.backend.getCustomerInfo(appUserID: appUserID,
+                                     withRandomDelay: isAppBackgrounded,
+                                     allowComputingOffline: allowComputingOffline,
+                                     completion: completion)
+    }
+
+    // Note: this is just a best guess.
+    private static let sourceForUnfinishedTransaction: PurchaseSource = .init(
+        isRestore: false,
+        // This might have been in theory a `.purchase`. The only downside of this is that the server
+        // won't validate that the product is present in the receipt.
```

Comment:
Hmm can you explain more in what situation this transaction could originate from a `purchase`? If we are in the middle of posting the transaction after a purchase, this should wait for that I guess?

Reply:
1. Server offline
2. Purchase fails
3. Compute customer info (but transaction still in queue)
4. Next time you launch the app, server is back up
5. Pending transaction -> post receipt

Technically that _was_ a `.purchase` when it was done originally, but we're sending it as `.queue` now.

Reply:
Ahh right, that would also happen currently if a POST receipt fails and it's retried later on right? But yeah, that makes sense 👍